File: android/src/main/java/com/shakebugs/react/utils/Files.java
package com.shakebugs.react.utils;

class Files {
    public static String removeExtension(String fileName) {
        if (fileName == null) {
            return null;
        }

        int index = fileName.lastIndexOf('.');
        if (index == -1 || index == 0) {
            return fileName;
        } else {
            return fileName.substring(0, index);
        }
    }
}

File: android/src/main/java/com/shakebugs/react/utils/Files.kt
package com.shakebugs.react.utils

internal object Files {
    @JvmStatic
    fun removeExtension(fileName: String): String {
        val index = fileName.lastIndexOf('.')
        return if (index == -1 || index == 0) {
            fileName
        } else {
            fileName.substring(0, index)
        }
    }
}
